### DISCUSSION TITLE
.Net MAUI XAML controls Memory Consumption

### DISCUSSION THREAD

**Discussion by DevashriK** [community/community#125584] [Programming Help]
.NET MAUI memory leaks

Bug

### Body

Hi All, 

I have a .Net MAUI desktop application which has a list of controls like .Net MAUI list view, Scroll Viewer, Collection Views, Toolkit Popups and syncfusion controls like SfDataGrid, SFComboBox, SfTextInputLayout, SFTabView etc. Whenever my app memory usage exceeds 750MB with Power usage as Very High, it becomes very slow and becomes unresponsive in around half hour. Also it suddenly crashes with errors like "The text associated with this error code could not be found. Layout cycle detected.  Layout could not complete."  and Access Violation errors. Any suggestions how to release the memory? I have tried releasing the memory consumed by Observable collections , clearing viewmodel objects and releasing the memory consumed by content views. But still the resources are not getting released. Please suggest how I can handle this.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @DevashriK ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by davevad93** [comment]
Hi @DevashriK ,

Personally I've only developed `WPF` apps and I have no experience with `MAUI`, but I'm gonna leave some resources about memory leaks and performance improvements in MAUI that might help you:

1. [Memory-Leaks](https://github.com/dotnet/maui/wiki/Memory-Leaks)
2. [Improve app performance](https://learn.microsoft.com/en-us/dotnet/maui/deployment/performance?view=net-maui-8.0)
3. [MemoryToolkit.Maui](https://github.com/AdamEssenmacher/MemoryToolkit.Maui)
